### ISSUE TITLE
org.h2.mvstore.Chunk.accountForRemovedPage NPE

### ISSUE THREAD

**Issue by Doc1faux** [nitrite/nitrite-java#867]
Hi, I'm currently using Nitrite `v3.4.2` and I got a crash which seems tied to [h2database](https://github.com/h2database/h2database) `v1.4.200`, I found no trace of NO2 in it...
Here is the stacktrace:
```
Fatal Exception: java.lang.IllegalStateException
java.lang.NullPointerException: Attempt to invoke virtual method 'boolean org.h2.mvstore.Chunk.accountForRemovedPage(int, boolean, long, long)' on a null object reference [1.4.200/3]
org.h2.mvstore.DataUtils.newIllegalStateException (DataUtils.java:950)
org.h2.mvstore.MVStore.store (MVStore.java:1338)
org.h2.mvstore.MVStore.store (MVStore.java:1312)
org.h2.mvstore.MVStore.tryCommit (MVStore.java:1272)
org.h2.mvstore.MVStore.writeInBackground (MVStore.java:2780)
org.h2.mvstore.MVStore$BackgroundWriterThread.run (MVStore.java:3290)
Caused by java.lang.NullPointerException
Attempt to invoke virtual method 'boolean org.h2.mvstore.Chunk.accountForRemovedPage(int, boolean, long, long)' on a null object reference
org.h2.mvstore.MVStore.acceptChunkOccupancyChanges (MVStore.java:1612)
org.h2.mvstore.MVStore.storeNow (MVStore.java:1433)
org.h2.mvstore.MVStore.store (MVStore.java:1334)
org.h2.mvstore.MVStore.store (MVStore.java:1312)
org.h2.mvstore.MVStore.tryCommit (MVStore.java:1272)
org.h2.mvstore.MVStore.writeInBackground (MVStore.java:2780)
org.h2.mvstore.MVStore$BackgroundWriterThread.run (MVStore.java:3290)
```

Similar issues have already been reported on their project:

- https://github.com/h2database/h2database/issues/2506 (closed for duplicate)
- https://github.com/h2database/h2database/issues/2436

The issue seems still opened on their side.
I've seen you've upgraded the dependency in the `v4.x` version to `v2.1.214`, maybe this would fix it?

For the moment, I got the crash only for a user 2 times, so it's not urgent.

**Comment by anidotnet** [comment]
Hope this is fixed in mvstore v2.1.214. Have not encountered any issue yet with this version in Nitrite 4.x.

**Comment by anidotnet** [comment]
Nitrite 4.2.0 has been released to maven central. Please read the [guide](https://nitrite.dizitart.com/java-sdk/getting-started/index.html) before upgrading from 3.x
